docs/developer-docs/latest/developer-resources/database-apis-reference/graphql-api.md

```diff
@@ -0,0 +1,471 @@
+:::
+
+## Unified response format
+
+Whatever the query, the response can contain the following fields:
+
+- `data`: the response data itself, which could be:
+  - a single entry, as an object with the following keys:
+    - `id` (number)
+    - `attributes` (object)
+    - `meta` (object)
+  - a list of entries, as an array of objects
+  - a custom response
+
+- `meta`(object): information about pagination, publication state, available locales…
+<!-- TODO: create an entry in the docs to list all errors -->
+
+## Queries
+
+We assume that the [Shadow CRUD](/developer-docs/latest/development/plugins/graphql.md#shadow-crud) feature is enabled. For each model, the GraphQL plugin auto-generates queries and mutations which just fit to your needs.
+
+### Fetch a single entry
+
+Single entries can be found by their `id`.
+
+:::request Example query: Find the entry with id 1, only the French fields
+```graphql
+query {
+  document(id: 1, locale: "fr", publicationState: LIVE) {
+    data {
+      id
+      attributes {
+        title
+        locale
+        categories {
+          data {
+            id
+            attributes {
+              name
+            }
+          }
+          meta {
+            pagination {
+              page
+              pageSize
+              total
+              pageCount
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+
+:::
+
+### Fetch multiple entries
+
+:::request Example query: Find all documents and populate 'categories' relation with the `name` attribute
+
+```graphql
+query {
+  documents {
+    data {
+      id
+      attributes {
+        title
+        locale
+        categories {
+          data {
+            id
+            attributes {
+                name
+            }
+          }
+        }
+      }
+    }
+    meta {
+      pagination {
+        page
+        pageSize
+        total
+        pageCount
+      }
+    }
+  }
+}
+
+```
+
+:::
+
+<!-- ? is this part 👇 still relevant? -->
+<!-- ### Fetch dynamic zone data
+
+Dynamic zones are union types in graphql so you need to use fragments to query the fields.
+
+:::request Query
+```graphql
+query {
+  restaurants {
+    dz {
+      __typename
+      ... on ComponentDefaultClosingperiod {
+        label
+      }
+    }
+  }
+}
+```
+::: -->
+
+## Mutations
+
+| Argument     | Type   | Description      |
+| ------------ | ------ | ---------------- |
+| `id`         | String | Entry id         |
+| `attributes` | Object | Entry attributes |
+
+### Create a new entry
+
+:::request Mutation
+
+```graphql
+mutation createArticle {
+  createArticle(data: { title: "Hello", relation: 1 }) {
+    data {
+      id
+      attributes {
+        title
+      }
+    }
+  }
+}
+```
+
+:::
+
+<!-- ? is it still relevant 👇 ? -->
+<!-- The implementation of the mutations also supports relational attributes. For example, you can create a new `User` and attach many `Restaurant` to it by writing your query like this:
```

Comment:
I think we will even allow creating those restaurants on demand and not just linking to existing but need to confirm with @Convly

Reply:
We can still reference relations to link when creating an entry, but I'm wondering if the following example is relevant since you're already creating an entry with an associated relation in the example above.
(+ in the current example, the IDs are in Mongo format, we would need to change that if we re-use it)